packages/pages/src/common/src/feature/stream.ts

```diff
@@ -75,3 +115,54 @@ export const convertRedirectConfigToStreamConfig = (
     };
   }
 };
+
+/**
+ * Loads the site stream specified in config.yaml or site-stream.json into a {@link SiteStream}.
+ */
+export const readSiteStream = (
+  projectStructure: ProjectStructure
+): SiteStream | undefined => {
+  // read site stream from deprecated sites-config directory if it exists
+  const siteStreamJsonPath = path.resolve(
+    projectStructure.getSitesConfigPath().path,
+    projectStructure.config.sitesConfigFiles.siteStream
+  );
+  if (fs.existsSync(siteStreamJsonPath)) {
+    const sitesJson = readJsonSync(siteStreamJsonPath);
+    return formatSiteStream(sitesJson, siteStreamJsonPath);
+  }
+
+  // read site stream from config.yaml
+  const configYamlPath = projectStructure.getConfigYamlPath().getAbsolutePath();
+  if (fs.existsSync(configYamlPath)) {
+    const yamlDoc = YAML.parse(fs.readFileSync(configYamlPath, "utf-8"));
+    if (yamlDoc.siteStream) {
+      yamlDoc.siteStream.entityId = yamlDoc.siteStream?.entityId?.toString();
+      return yamlDoc.siteStream;
+    }
+  }
+
+  return;
+};
+
+/**
+ * Converts the deprecated format of a siteStream specified in site-stream.json into
+ * the format of a siteStream specified in config.yaml
+ */
+export const formatSiteStream = (sitesJson: any, siteStreamPath: string) => {
```

Comment:
Can we create a type for the site-stream.json format?

Reply:
That should just be the Stream type right?

Also this formatter supported site stream $id's that were not "site-stream", is there any situation where that would be the case?